### PR TITLE
feat: Add ArgoCD plugin to service catalog

### LINKED ISSUE #20: Plugin: Install and configure ArgoCD plugin
Show ArgoCD status in Backstage for componetns

https://roadie.io/backstage/plugins/argo-cd/?utm_source=backstage.io&utm_medium=marketplace&utm_campaign=argo-cd

### PR DESCRIPTION
**Pull request by mimotej**
This PR adds ArgoCD plugin to the service page. Plugin shows name, sync status, health and when it was last synced:
image

Resolves #20

### CODE REVIEW

**Review by tumido — CHANGES_REQUESTED** [review]
Couple of changes required. I suspect the deployment manifests were not tested because they are not kubernetes compatible. :stuck_out_tongue:  See inline comments.

File: app-config.yaml [hunk 1]
```diff
@@ -44,6 +44,17 @@ backend:
     # Use this to configure a working directory for the scaffolder, defaults to the OS temp-dir
     # workingDirectory: /tmp
 
+proxy:
+  '/argocd/api':
+    # url to the api of your hosted argoCD instance
+    target: https://argocd.operate-first.cloud/api/v1/
+    changeOrigin: true
+    # this line is required if your hosted argoCD instance has self-signed certificate
+    secure: false
```
[the remark this answers is not in the document]
Comment (mimotej) [reply app-config.yaml]: Resolved in latest commit.

File: manifests/base/deployment.yaml [hunk 2]
```diff
@@ -28,6 +28,10 @@ spec:
                 name: service-catalog-pguser-service-catalog
             - secretRef:
                 name: service-catalog
+                key: BACKEND_SECRET
+            - secretRef:
+                name: service-catalog
+                key: ARGOCD_AUTH_TOKEN
```
[the remark this answers is not in the document]
Comment (mimotej) [reply manifests/base/deployment.yaml]: Resolved in latest commit.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**